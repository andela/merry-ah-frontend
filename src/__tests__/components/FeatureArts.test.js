import React from 'react';
import { mount } from 'enzyme';
import props from '../__mocks__/__featureMocks__/propsMock';
import {
  FeaturedArts,
} from '../../components/FeaturedArts.jsx';


let wrapper;
beforeEach(() => {
  wrapper = mount(
    <FeaturedArts {...props} />,
  );
});
describe('FeaturedArts Component', () => {
  it('should match snapshot', () => {
    wrapper.instance().componentDidMount();
    expect(wrapper).toMatchSnapshot();
  });
  it('should have 3 feature cards', () => {
    expect(wrapper.find('FeatureCard').length).toEqual(3);
  });
  it('should have 2 Arrow components', () => {
    expect(wrapper.find('Arrow').length).toEqual(2);
  });
  it('should have one active card', () => {
    expect(wrapper.find('.cards_active').length).toEqual(1);
  });
  it('should have two inactive card', () => {
    expect(wrapper.find('.cards_inactive').length).toEqual(2);
  });
  it('should have three feature buttons', () => {
    expect(wrapper.find('FeatureButton').length).toEqual(3);
  });
  it('should have view as text for Featurebutton', () => {
    expect(wrapper
      .find('FeatureButton').first().render().text()).toContain('View');
  });
  it('should have props for imgUrl', () => {
    expect(wrapper
      .find('FeatureCard')
      .first().props().imgUrl).toContain('../assets/images/1.jpg');
  });
  it('should have props for title', () => {
    expect(wrapper
      .find('FeatureCard')
      .first().props().title).toContain('Do you love art');
  });
});
