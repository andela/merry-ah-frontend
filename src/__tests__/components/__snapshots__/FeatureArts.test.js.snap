// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedArts Component should match snapshot 1`] = `
<FeaturedArts
  activeList={
    Array [
      Object {
        "category": "painting",
        "objectID": 1,
        "title": "Do you love art",
        "url": "../assets/images/1.jpg",
      },
      Object {
        "category": "freehand_sketch",
        "objectID": 2,
        "title": "Cirrhosis of the sky",
        "url": "../assets/images/2.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 3,
        "title": "Walkers Short Bread",
        "url": "../assets/images/3.jpg",
      },
    ]
  }
  artList={
    Array [
      Object {
        "category": "painting",
        "objectID": 1,
        "title": "Do you love art",
        "url": "../assets/images/1.jpg",
      },
      Object {
        "category": "freehand_sketch",
        "objectID": 2,
        "title": "Cirrhosis of the sky",
        "url": "../assets/images/2.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 3,
        "title": "Walkers Short Bread",
        "url": "../assets/images/3.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 4,
        "title": "How to have a blast",
        "url": "../assets/images/4.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 5,
        "title": "Debugging best practices",
        "url": "../assets/images/5.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 6,
        "title": "Fast and furious remake",
        "url": "../assets/images/6.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 7,
        "title": "Best Coder alive",
        "url": "../assets/images/7.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 8,
        "title": "Fifa e-world cup",
        "url": "../assets/images/8.jpg",
      },
      Object {
        "category": "mosaic",
        "objectID": 9,
        "title": "Perfection at its peak",
        "url": "../assets/images/9.jpg",
      },
    ]
  }
  data={
    Object {
      "activeList": Array [
        Object {
          "category": "painting",
          "objectID": 1,
          "title": "Do you love art",
          "url": "../assets/images/1.jpg",
        },
        Object {
          "category": "freehand_sketch",
          "objectID": 2,
          "title": "Cirrhosis of the sky",
          "url": "../assets/images/2.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 3,
          "title": "Walkers Short Bread",
          "url": "../assets/images/3.jpg",
        },
      ],
      "artList": Array [
        Object {
          "category": "painting",
          "objectID": 1,
          "title": "Do you love art",
          "url": "../assets/images/1.jpg",
        },
        Object {
          "category": "freehand_sketch",
          "objectID": 2,
          "title": "Cirrhosis of the sky",
          "url": "../assets/images/2.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 3,
          "title": "Walkers Short Bread",
          "url": "../assets/images/3.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 4,
          "title": "How to have a blast",
          "url": "../assets/images/4.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 5,
          "title": "Debugging best practices",
          "url": "../assets/images/5.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 6,
          "title": "Fast and furious remake",
          "url": "../assets/images/6.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 7,
          "title": "Best Coder alive",
          "url": "../assets/images/7.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 8,
          "title": "Fifa e-world cup",
          "url": "../assets/images/8.jpg",
        },
        Object {
          "category": "mosaic",
          "objectID": 9,
          "title": "Perfection at its peak",
          "url": "../assets/images/9.jpg",
        },
      ],
    }
  }
  nextSlide={[Function]}
  previousSlide={[Function]}
  setInitData={[Function]}
>
  <div>
    <div
      id="feature_list"
    >
      <div
        id="featured_arts_header"
      >
        <h5>
          Featured Arts
        </h5>
        <hr />
      </div>
      <FeatureCard
        activeClass="cards_inactive"
        imgUrl="../assets/images/1.jpg"
        key="1"
        title="Do you love art"
      >
        <div
          className="cards_inactive"
          id="cards"
          style={
            Object {
              "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                        rgba(0, 0, 0, 0.5)), url(../assets/images/1.jpg) no-repeat center ",
              "backgroundSize": "cover",
            }
          }
        >
          <div
            id="feature_button_div"
          >
            <h5>
              Do you love art
            </h5>
            <FeatureButton
              myclass="feature_button_inactive"
              text="View"
            >
              <button
                className="feature_button_inactive"
              >
                View
              </button>
            </FeatureButton>
          </div>
        </div>
      </FeatureCard>
      <FeatureCard
        activeClass="cards_active"
        imgUrl="../assets/images/2.jpg"
        key="2"
        title="Cirrhosis of the sky"
      >
        <div
          className="cards_active"
          id="cards"
          style={
            Object {
              "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                        rgba(0, 0, 0, 0.5)), url(../assets/images/2.jpg) no-repeat center ",
              "backgroundSize": "cover",
            }
          }
        >
          <div
            id="feature_button_div"
          >
            <h5>
              Cirrhosis of the sky
            </h5>
            <FeatureButton
              myclass="feature_button"
              text="View"
            >
              <button
                className="feature_button"
              >
                View
              </button>
            </FeatureButton>
          </div>
        </div>
      </FeatureCard>
      <FeatureCard
        activeClass="cards_inactive"
        imgUrl="../assets/images/3.jpg"
        key="3"
        title="Walkers Short Bread"
      >
        <div
          className="cards_inactive"
          id="cards"
          style={
            Object {
              "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                        rgba(0, 0, 0, 0.5)), url(../assets/images/3.jpg) no-repeat center ",
              "backgroundSize": "cover",
            }
          }
        >
          <div
            id="feature_button_div"
          >
            <h5>
              Walkers Short Bread
            </h5>
            <FeatureButton
              myclass="feature_button_inactive"
              text="View"
            >
              <button
                className="feature_button_inactive"
              >
                View
              </button>
            </FeatureButton>
          </div>
        </div>
      </FeatureCard>
      <div
        id="carousel_control"
      >
        <Arrow
          className="slide-arrow"
          clickFunction={[Function]}
          direction="left"
          glyph="◀"
        >
          <div
            className="slide-arrow left"
            onClick={[Function]}
          >
            ◀
          </div>
        </Arrow>
        <Arrow
          className="slide-arrow"
          clickFunction={[Function]}
          direction="right"
          glyph="▶"
        >
          <div
            className="slide-arrow right"
            onClick={[Function]}
          >
            ▶
          </div>
        </Arrow>
      </div>
    </div>
  </div>
</FeaturedArts>
`;
